### PR TITLE
Create sbom-utility-scripts image

### PR DESCRIPTION
**Pull request by mkosiarc**
This image contains multiple python scripts that are used in the buildah tasks. This is done so that multiple steps in buildah tasks can be merged. The image contains scripts from the following stages:

merge-syft-sboms
merge-cachi2-sbom
create-purl-sbom
create-base-images-sbom

NOTE: the "merge_syft_sbom" from the "merge-cachi2-sbom" stage was taken from
https://github.com/containerbuildsystem/cachi2/blob/main/utils/merge_syft_sbom.py and renamed to "merge_cachi2_sboms.py". It will be removed from the cachi2 repo and image.

STONEBLD-2608

### CODE REVIEW

**Review by chmeliik — APPROVED** [review]

File: .github/workflows/build-sbom-utility-scripts-image.yml [hunk 1]
```diff
@@ -31,10 +31,16 @@ jobs:
...
       run: |
         python3 -m pip install tox
-        cd ./base-images-sbom-script/app/
+        cd ./sbom-utility-scripts/scripts/base-images-sbom-script/app/
+        tox
+     
+    - name: Run tox checks for base-images-sbom-script
+      run: |
+        python3 -m pip install tox
+        cd ./sbom-utility-scripts/scripts/merge-cachi2-sboms/
```
Comment (chmeliik) [inline .github/workflows/build-sbom-utility-scripts-image.yml:43]:
```suggestion
        cd ./sbom-utility-scripts/scripts/merge-cachi2-sboms-script/
```

**Comment by mkosiarc** [comment]
Associated PR https://github.com/konflux-ci/build-definitions/pull/1291

**Review by tkdchen — COMMENTED** [review]
LGTM overall. A suggestion is to make the input and output files, these scripts operates on, as command line arguments. Then, the SBOM merge flow will be clearer for easier understanding by specifying data exchange files explicitly.

**Comment by mkosiarc** [comment]
> LGTM overall. A suggestion is to make the input and output files, these scripts operates on, as command line arguments. Then, the SBOM merge flow will be clearer for easier understanding by specifying data exchange files explicitly.

Adam raised the same issue in https://github.com/konflux-ci/build-definitions/pull/1291#discussion_r1715417017
I agree with that, but I would like to do in next iteration.

**Review by mmorhun — APPROVED** [review]

**Pull request merged**